From 98b6967433b7f9f100e106ebd413986d4bcb9910 Mon Sep 17 00:00:00 2001
From: Yael Samet <yael.samet@intel.com>
Date: Sun, 6 Nov 2016 08:36:33 +0200
Subject: [PATCH 4640/4706] mei: dal: prevent exclusive access to the spooler
 applet.

commit 759bc62bcccd7ab013cd4a2f6f43c4dfaee29176 from
git://git.yoctoproject.org/linux-yocto-4.1

The spooler applet handles all other applets and cannot
be owned by anybody.

Change-Id: I1b82779afcda98f99a19e704eda37f91091f7243
Signed-off-by: Yael Samet <yael.samet@intel.com>
Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
---
 drivers/misc/mei/dal/dal_ta_access.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/misc/mei/dal/dal_ta_access.c b/drivers/misc/mei/dal/dal_ta_access.c
index 1b31ce8..2925ecc 100644
--- a/drivers/misc/mei/dal/dal_ta_access.c
+++ b/drivers/misc/mei/dal/dal_ta_access.c
@@ -64,6 +64,12 @@
 #include <linux/mei_cl_bus.h>
 #include "dal_dev.h"
 
+/* Spooler UUID */
+#define SPOOLER_UUID UUID_BE(0xba8d1643, 0x50b6, 0x49cc, \
+			     0x86, 0x1d, 0x2c, 0x01, 0xbe, 0xd1, 0x4b, 0xe8)
+
+static const uuid_be spooler_ta_id = SPOOLER_UUID;
+
 struct dal_access_policy {
 	struct list_head list;
 	uuid_be app_id;
@@ -186,5 +192,8 @@ int dal_access_list_init(struct dal_device *ddev)
 	INIT_LIST_HEAD(access_list);
 	dev_set_drvdata(&ddev->dev, access_list);
 
+	/* Nobody can own SPOOLER TA */
+	dal_access_policy_add(ddev, spooler_ta_id, NULL);
+
 	return 0;
 }
-- 
1.7.5.4

